Latest bap does not compile on 32bit machine (x86 / i686)

```
opam repository add git://github.com/BinaryAnalysisPlatform/opam-repository.git
opam update
opam install bap
```
This results in:
```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of bap-cache failed at "make".
Processing  1/2: [bap-cache: ocamlfind remove]
#=== ERROR while installing bap-cache.1.0.0~alpha =============================#
# opam-version 1.2.2
# os           linux
# command      make
# path         /home/myuser/.opam/4.02.3/build/bap-cache.1.0.0~alpha
# compiler     4.02.3
# exit-code    2
# env-file     /home/myuser/.opam/4.02.3/build/bap-cache.1.0.0~alpha/bap-cache-1050-d2d111.env
# stdout-file  /home/myuser/.opam/4.02.3/build/bap-cache.1.0.0~alpha/bap-cache-1050-d2d111.out
# stderr-file  /home/myuser/.opam/4.02.3/build/bap-cache.1.0.0~alpha/bap-cache-1050-d2d111.err
### stdout ###
# ocaml setup.ml -quiet -build 
# ocamlfind ocamlopt -package unix -package ocamlbuild -linkpkg -package core_kernel -package findlib -package ppx_driver.ocamlbuild myocamlbuild.ml /home/myuser/.opam/4.02.3/lib/ocaml/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
# /home/myuser/.opam/4.02.3/bin/ocamlfind ocamldep -package bap -package cmdliner -package core_kernel -package ppx_jane -pp 'ppx-jane -dump-ast -inline-test-lib bap' -predicates ppx_driver -modules plugins/cache/cache_main.ml > plugins/cache/cache_main.ml.depends
# /home/myuser/.opam/4.02.3/bin/ocamlfind ocamlc -c -g -annot -bin-annot -short-paths -package bap -package cmdliner -package core_kernel -package ppx_jane -pp 'ppx-jane -dump-ast -inline-test-lib bap' -predicates ppx_driver -I plugins/cache -o plugins/cache/cache_main.cmo plugins/cache/cache_main.ml
# + /home/myuser/.opam/4.02.3/bin/ocamlfind ocamlc -c -g -annot -bin-annot -short-paths -package bap -package cmdliner -package core_kernel -package ppx_jane -pp 'ppx-jane -dump-ast -inline-test-lib bap' -predicates ppx_driver -I plugins/cache -o plugins/cache/cache_main.cmo plugins/cache/cache_main.ml
# File "plugins/cache/cache_main.ml", line 31, characters 15-28:
# Error: Integer literal exceeds the range of representable integers of type int
# Command exited with code 2.
### stderr ###
# make: *** [build] Error 1



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
```

--- a/plugins/cache/cache_main.ml
+++ b/plugins/cache/cache_main.ml
@@ -10,11 +10,11 @@ type entry = {
   ctime   : float;
   hits    : int;
   path    : string;
-  size    : int;
+  size    : int64;
 } [@@deriving bin_io, compare, sexp]
 
 type config = {
-  max_size : int;
+  max_size : int64;
 } [@@deriving bin_io, compare, sexp]
 
 type index = {
@@ -28,7 +28,7 @@ module Index = struct
   let index_file = "index"
   let lock_file = "lock"
   let default_config = {
-    max_size = 5_000_000_000;
+    max_size = 5_000_000_000L;
   }
   let empty = {
     config = default_config;
@@ -59,8 +59,8 @@ module Index = struct
     cache_dir
 
   let size idx =
-    Map.fold idx.entries ~init:0 ~f:(fun ~key:_ ~data:e size ->
-        size + e.size)
+    Map.fold idx.entries ~init:0L ~f:(fun ~key:_ ~data:e size ->
+        Int64.(size + e.size))
 
   (* freq is a frequence of accesses to the cache entry in
      access per second. *)
@@ -77,7 +77,7 @@ module Index = struct
 
   let rec clean idx =
     let size = size idx in
-    if size > 0 && size > idx.config.max_size
+    if size > 0L && size > idx.config.max_size
     then clean (evict_entry idx)
     else idx
 
@@ -127,7 +127,7 @@ module Index = struct
 end
 
 let size file =
-  Unix.((stat file).st_size)
+  Unix.LargeFile.((stat file).st_size)
 
 let cleanup () =
   Index.update ~f:(fun _ idx ->
@@ -136,14 +136,14 @@ let cleanup () =
 
 let set_size size =
   Index.update ~f:(fun _ idx ->
-      {idx with config = {max_size = size * 1024 * 1024}});
+      {idx with config = {max_size = Int64.(size * 1024L * 1024L)}});
   exit 0
 
 let print_info () =
   Index.run ~f:(fun idx ->
-      let mb s = Int.(s / 1024 / 1024) in
-      printf "Maximum size: %5d MB@\n" @@ mb idx.config.max_size;
-      printf "Current size: %5d MB@\n" @@ mb (Index.size idx));
+      let mb s = Int64.(s / 1024L / 1024L) in
+      printf "Maximum size: %5Ld MB@\n" @@ mb idx.config.max_size;
+      printf "Current size: %5Ld MB@\n" @@ mb (Index.size idx));
   exit 0
 
 let set_dir dir = match dir with
@@ -198,7 +198,7 @@ let set_size =
   let doc =
     "Set maximum total size of cached data to $(docv) MB. The
   option value will persist between different runs of the program" in
-  Arg.(value & opt (some int) None & info ["size"] ~doc ~docv:"N")
+  Arg.(value & opt (some int64) None & info ["size"] ~doc ~docv:"N")
 
 let dir =
   let doc = "Use $(docv) as a cache directory" in

